design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,190 @@
+# Rook-Ceph Bucket Provisioning
+
+## Proposed Feature
+
+Add a CephObjectBucketClaim API endpoint and control loop to the Rook-Ceph operator for the dynamic provisioning of S3 buckets by Kubernetes users.
+
+### Overview
+
+  Rook-Ceph has developed Custom Resource Definitions (CRDs) to support the dynamic provisioning of Ceph Object Stores and Ceph Object Store Users.  This proposal introduces the next logical step: dynamic bucket provisioning with the addition of a CephObjectBucketClaim CRD.  The intent is to round out the Rook-Ceph experience for Kubernetes users by providing a single control plane for the managing of Ceph Object components.      
+
+## Goals
+
+- Provide Rook-Ceph users the ability to dynamically provision object buckets via the Kubernetes API with a CephObjectBucketClaim CRD.
+- Enable cluster admin control over bucket creation and object store access via the Kubernetes API.
+- Utilize a familiar and deterministic pattern when injecting bucket connection information into workload environments. 
+
+## Non-Goals
+
+- This proposal does not implement a generalized method for bucket provisioning; it will be specific to Rook-Ceph
+- This design does not attempt to implement a 1:1 model of Kubernetes PersistentVolumes and PersistentVolumeClaims.  Some shallow similarities will exist. 
+- This design does not include the Swift interface implementation by Ceph Object.
+- This design does not provide users a means of deleting buckets via the Kubernetes API in order to avoid accidental loss of data.  A Delete operation only deletes the Kubernetes API objects.
+
```

Comment:
Another non-goal is probably that this design does not address bucket quotas. It is expected that Kubernetes CRD quotas will handle this.